cudapoa/src/cudapoa_nw_tb_banded.cuh

```diff
@@ -0,0 +1,586 @@
+                            {
+                                int32_t trace_tmp = -(score_gIdx - pred_idx_tmp);
+                                int32_t score_tmp = get_score_tb(scores, pred_idx_tmp, -1, score_matrix_height, gradient, band_width, max_column, min_score_value);
+                                if (penalty < score_tmp)
+                                {
+                                    penalty          = score_tmp;
+                                    traceback[index] = trace_tmp;
+                                }
+                            }
+                        }
+                        first_element_prev_score = penalty + gap_score;
+                        set_score_tb(scores, score_gIdx, -1, score_matrix_height, first_element_prev_score, gradient, band_width, max_column);
+                    }
+                }
+                else
+                {
+                    penalty = min_score_value;
+                    // pick the predecessor which falls within the limit of score_matrix_height
+                    for (int32_t p = 1; p < pred_count; p++)
+                    {
+                        pred_node_id         = incoming_edges[node_id * CUDAPOA_MAX_NODE_EDGES + p];
+                        int32_t pred_idx_tmp = node_id_to_pos[pred_node_id] + 1;
+                        if ((score_gIdx - pred_idx_tmp) < score_matrix_height)
+                        {
+                            int32_t trace_tmp = -(score_gIdx - pred_idx_tmp);
+                            int32_t score_tmp = get_score_tb(scores, pred_idx_tmp, -1, score_matrix_height, gradient, band_width, max_column, min_score_value);
+                            if (penalty < score_tmp)
+                            {
+                                penalty          = score_tmp;
+                                traceback[index] = trace_tmp;
+                            }
+                        }
+                    }
+                    first_element_prev_score = penalty + gap_score;
+                    set_score_tb(scores, score_gIdx, -1, score_matrix_height, first_element_prev_score, gradient, band_width, max_column);
+                }
+            }
+        }
+        pred_count = __shfl_sync(FULL_MASK, pred_count, 0);
+        pred_idx   = __shfl_sync(FULL_MASK, pred_idx, 0);
+        //-------------------------------------------------------------
+
+        SeqT graph_base = nodes[node_id];
+
+        for (int32_t read_pos = lane_idx * CELLS_PER_THREAD + band_start; read_pos < band_start + band_width; read_pos += CUDAPOA_MIN_BAND_WIDTH)
+        {
+            SeqT4<SeqT>* d_read4 = (SeqT4<SeqT>*)read;
+            SeqT4<SeqT> read4    = d_read4[read_pos / CELLS_PER_THREAD];
+
+            TraceT4<TraceT> trace;
+            ScoreT4<ScoreT> score = {min_score_value, min_score_value, min_score_value, min_score_value};
+            get_scores_tb(scores, pred_idx, score_gIdx, read_pos, score_matrix_height, gradient, band_width, max_column,
```

Comment:
can you add a comment here explaining how the score_matrix_height parameter is used?

Reply:
done